### PR TITLE
[WCF] Modernize and refactor tests

### PR DESCRIPTION
**Pull request by matt-hensley**
## Changes

WCF tests updated to use modern C# features, share constants, and common assertions extracted.

Tests should be functionally identical.

This is in preparation for updating to latest RPC semconv.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Wcf/Implementation/WcfInstrumentationConstants.cs
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.Wcf/Implementation/WcfInstrumentationConstants.cs]: worth to introduce same naming convention a in shared SemanticConvention file.

```suggestion
public const string AttributeSoapMessageVersion = "soap.message_version";
public const string AttributeSoapReplyAction = "soap.reply_action";
public const string AttributeSoapVia = "soap.via";
public const string AttributeWcfChannelScheme = "wcf.channel.scheme";
public const string AttributeWcfChannelPathT = "wcf.channel.path";
```

File: test/OpenTelemetry.Instrumentation.Wcf.Tests/TelemetryDispatchMessageInspectorTests.netfx.cs [hunk 1]
```diff
@@ -114,8 +114,6 @@ public async Task IncomingRequestInstrumentationTest(
                                     case WcfEnrichEventNames.BeforeSendReply:
                                         activity.SetTag("server.beforesendreply", WcfEnrichEventNames.BeforeSendReply);
                                         break;
-                                    default:
-                                        break;
                                 }
```
Comment (Kielek) [inline test/OpenTelemetry.Instrumentation.Wcf.Tests/TelemetryDispatchMessageInspectorTests.netfx.cs:116]: Something change in the C# analyzers. I am pretty sure that I was forced to add empty defaults, could you please double check it?

Comment (matt-hensley) [reply test/OpenTelemetry.Instrumentation.Wcf.Tests/TelemetryDispatchMessageInspectorTests.netfx.cs:116]: I'm surprised the exhaustive case check isn't failing. Probably an edge case or bug.

I put the default case back.

**Review by Kielek — APPROVED** [review]
